modules/misc/mozilla-messaging-hosts.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs.stdenv) isDarwin;
+
+  cfg = config.mozilla;
+
+  defaultPaths = [
+    # Link a .keep file to keep the directory around
+    (pkgs.writeTextDir "lib/mozilla/native-messaging-hosts/.keep" "")
+  ];
+
+  thunderbirdNativeMessagingHostsPath = if isDarwin then
+    "Library/Mozilla/NativeMessagingHosts"
+  else
+    ".mozilla/native-messaging-hosts";
+
+  firefoxNativeMessagingHostsPath = if isDarwin then
+    "Library/Application Support/Mozilla/NativeMessagingHosts"
+  else
+    ".mozilla/native-messaging-hosts";
+in {
+  meta.maintainers =
+    [ maintainers.booxter maintainers.rycee hm.maintainers.bricked ];
+
+  options.mozilla = {
+    firefoxNativeMessagingHosts = mkOption {
+      internal = true;
+      type = with types; listOf package;
+      default = [ ];
+      description = ''
+        List of Firefox native messaging hosts to configure.
+      '';
+    };
+
+    thunderbirdNativeMessagingHosts = mkOption {
+      internal = true;
+      type = with types; listOf package;
+      default = [ ];
+      description = ''
+        List of Thunderbird native messaging hosts to configure.
+      '';
+    };
+  };
+
+  config = mkIf (cfg.firefoxNativeMessagingHosts != [ ]
+    || cfg.thunderbirdNativeMessagingHosts != [ ]) {
+      home.file = if isDarwin then
+        let
+          firefoxNativeMessagingHostsJoined = pkgs.symlinkJoin {
+            name = "ff-native-messaging-hosts";
+            paths = defaultPaths ++ cfg.firefoxNativeMessagingHosts;
+          };
+          thunderbirdNativeMessagingHostsJoined = pkgs.symlinkJoin {
```

Comment:
This will create the directory even when Thunderbird - or Firefox - is not enabled (the `||` conditional above should be adjusted).